### PR TITLE
Added wrap-doc methods #8

### PR DESCRIPTION
**Pull request by rahul799**
# Description

Please include a summary of the change and which issue is fixed.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/pyOpenMS/pxds/IsotopeModel.pxd [hunk 1]
```diff
@@ -5,17 +5,46 @@ from EmpiricalFormula cimport *
...
-        void setSamples(EmpiricalFormula &formula) nogil except +
+            # wrap-doc:
+                #   Set the offset of the model
+                #   -----
+                #   The whole model will be shifted to the new offset without being computing all over.
+                #   This leaves a discrepancy which is minor in small shifts (i.e. shifting by one or two
+                #   standard deviations) but can get significant otherwise. In that case use setParameters()
+                #   which enforces a recomputation of the model
+
+        double getOffset() nogil except + Get the offset of the model
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeModel.pxd:31]: Missing # wrap-doc

File: src/pyOpenMS/pxds/IsotopeModel.pxd [hunk 2]
```diff
@@ -5,17 +5,46 @@ from EmpiricalFormula cimport *
...
+        EmpiricalFormula getFormula() nogil except + Return the Averagine peptide formula (mass calculated from mean mass and charge -- use .setParameters() to set them)
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeModel.pxd:32]: Missing # wrap-doc

File: src/pyOpenMS/pxds/IsotopeModel.pxd [hunk 3]
```diff
@@ -5,17 +5,46 @@ from EmpiricalFormula cimport *
...
+        void setSamples(EmpiricalFormula &formula) nogil except + Set sample/supporting points of interpolation
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeModel.pxd:33]: Missing # wrap-doc

File: src/pyOpenMS/pxds/ItraqConstants.pxd [hunk 4]
```diff
@@ -24,7 +29,28 @@ cdef extern from "<OpenMS/ANALYSIS/QUANTITATION/ItraqConstants.h>" namespace "Op
         # double ISOTOPECORRECTIONS_TMT_SIXPLEX()
 
         StringList getIsotopeMatrixAsStringList(int itraq_type, libcpp_vector[Matrix[double] ] & isotope_corrections) nogil except +
+            # wrap-doc:
+                #   Convert isotope correction matrix to stringlist
+                #   -----
+                #   Each line is converted into a string of the format &lt;channel&gt;:&lt;-2Da&gt;/&lt;-1Da&gt;/&lt;+1Da&gt;/&lt;+2Da&gt; ; e.g. '114:0/0.3/4/0'
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/ItraqConstants.pxd:35]: I am not sure how that will be represented. Maybe reformat.

File: src/pyOpenMS/pxds/ItraqConstants.pxd [hunk 5]
```diff
@@ -24,7 +29,28 @@ cdef extern from "<OpenMS/ANALYSIS/QUANTITATION/ItraqConstants.h>" namespace "Op
...
+                #   Useful for creating parameters or debug output
+                #   -----
+                #   :param itraq_type: Which matrix to stringify. Should be of values from enum ITRAQ_TYPES
+                #   :param isotope_corrections: Vector of the two matrices (4plex, 8plex).
+
         void updateIsotopeMatrixFromStringList(int itraq_type, StringList & channels, libcpp_vector[Matrix[double] ] & isotope_corrections) nogil except +
+            # wrap-doc:
+                #   Convert strings to isotope correction matrix rows
+                #   -----
+                #   Each string of format &lt;channel&gt;:&lt;-2Da&gt;/&lt;-1Da&gt;/&lt;+1Da&gt;/&lt;+2Da&gt; ; e.g. '114:0/0.3/4/0'
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/ItraqConstants.pxd:45]: See comment above

File: src/pyOpenMS/pxds/KroenikFile.pxd [hunk 6]
```diff
@@ -8,8 +8,29 @@ from Feature cimport *
...
+                #   Every subsequent line is a feature
+                #   -----
+                #   All properties in the file are converted to Feature properties, whereas "First Scan", "Last Scan", "Num of Scans" and "Modifications" are stored as
+                #   metavalues with the following names "FirstScan", "LastScan", "NumOfScans" and "AveragineModifications"
+                #   -----
+                #   The width in m/z of the overall convex hull of each feature is set to 3 Th in lack of a value provided by the Kroenik file
 
         KroenikFile() nogil except +
 
         void store(String filename, MSSpectrum & spectrum)  nogil except +
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/KroenikFile.pxd:26]: e.g. Stores a MSExperiment into a Kroenik file

File: src/pyOpenMS/pxds/LocalLinearMap.pxd [hunk 7]
```diff
@@ -5,15 +5,29 @@ from Matrix cimport *
...
     cdef cppclass LocalLinearMap "OpenMS::LocalLinearMap":
+        # wrap-doc:
+            #   Trained Local Linear %Map (LLM) model for peak intensity prediction
+            #   -----
+            #   This class offers a model for predictions of peptide peak heights
+            #   (referred to as intensities) by a Local Linear %Map (LLM) model and
+            #   is the basis of PeakIntensityPredictor.
+            #   -----
+            #   A general introduction to the Peak Intensity Predictor (PIP)
+            #   can be found in the <A HREF="tutorial_pip.html">PIP Tutorial</A>.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LocalLinearMap.pxd:16]: The HREF is probably not working in the code either add a link to the tutorials, or just state PIP Tutorial.

File: src/pyOpenMS/pxds/LocalLinearMap.pxd [hunk 8]
```diff
@@ -5,15 +5,29 @@ from Matrix cimport *
 cdef extern from "<OpenMS/ANALYSIS/PIP/LocalLinearMap.h>" namespace "OpenMS":
     
     cdef cppclass LocalLinearMap "OpenMS::LocalLinearMap":
+        # wrap-doc:
+            #   Trained Local Linear %Map (LLM) model for peak intensity prediction
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LocalLinearMap.pxd:9]: Has the "%Map" a deeper meaning or can that be renamed to just "Map"

**Review by oliveralka — CHANGES_REQUESTED** [review]
Looks good, some minor changes!

File: src/pyOpenMS/pxds/LocalLinearMap.pxd (hunk 8 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/LocalLinearMap.pxd]: Fixed

File: src/pyOpenMS/pxds/IsotopeModel.pxd [hunk 9]
```diff
@@ -5,17 +5,46 @@ from EmpiricalFormula cimport *
 cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeModel.h>" namespace "OpenMS":
     
     cdef cppclass IsotopeModel "OpenMS::IsotopeModel":
+        # wrap-doc:
+                #   Isotope distribution approximated using linear interpolation
+                #   -----
+                #   This models a smoothed (widened) distribution, i.e. can be used to sample actual raw peaks (depending on the points you query).
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeModel.pxd:11]:
```suggestion
                #   This models a smoothed (widened) distribution, i.e. can be used to sample actual raw peaks (depending on the points queried)
```

File: src/pyOpenMS/pxds/IsotopeModel.pxd [hunk 10]
```diff
@@ -5,17 +5,46 @@ from EmpiricalFormula cimport *
...
         IsotopeModel(IsotopeModel &) nogil except +
         UInt getCharge() nogil except +
         void setOffset(double offset) nogil except +
-        double getOffset() nogil except +
-        EmpiricalFormula getFormula() nogil except +
-        void setSamples(EmpiricalFormula &formula) nogil except +
+            # wrap-doc:
+                #   Set the offset of the model
+                #   -----
+                #   The whole model will be shifted to the new offset without being computing all over.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeModel.pxd:26]:
```suggestion
                #   The whole model will be shifted to the new offset without being computing all over
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 11]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
-        void destroy() nogil except +
+        void destroy() nogil except + # wrap-doc:Deletes the singleton instance
         double getValueByMass(double t, double m, UInt z, Int mode) nogil except +
+            # wrap-doc:
+                #   Returns the value of the isotope wavelet at position `t`. Usually, you do not need to call this function
+                #   -----
+                #   Note that this functions returns the pure function value of psi and not the normalized (average=0)
+                #   value given by Psi
+                #   -----
+                #   :param t: The position at which the wavelet has to be drawn (within the coordinate system of the wavelet).
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:23]:
```suggestion
                #   :param t: The position at which the wavelet has to be drawn (within the coordinate system of the wavelet)
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 12]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   :param m: The m/z position within the signal (i.e. the mass not de-charged) within the signal.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:24]:
```suggestion
                #   :param m: The m/z position within the signal (i.e. the mass not de-charged) within the signal
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 13]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   :param z: The charge `z` we want to detect.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:25]:
```suggestion
                #   :param z: The charge `z` we want to detect
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 14]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   :param mode: Indicates whether positive mode (+1) or negative mode (-1) has been used for ionization.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:26]:
```suggestion
                #   :param mode: Indicates whether positive mode (+1) or negative mode (-1) has been used for ionization
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 15]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   :param t: The position at which the wavelet has to be drawn (within the coordinate system of the wavelet).
+                #   :param m: The m/z position within the signal (i.e. the mass not de-charged) within the signal.
+                #   :param z: The charge `z` we want to detect.
+                #   :param mode: Indicates whether positive mode (+1) or negative mode (-1) has been used for ionization. 
+
         double getValueByLambda(double lambda_, double tz1) nogil except +
+            # wrap-doc:
+                #   Returns the value of the isotope wavelet at position `t` via a fast table lookup
+                #   -----
+                #   Usually, you do not need to call this function.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:32]:
```suggestion
                #   Usually, you do not need to call this function
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 16]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   Please use `sampleTheWavelet` instead.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:33]:
```suggestion
                #   Please use `sampleTheWavelet` instead
```

File: src/pyOpenMS/pxds/IsotopeWavelet.pxd [hunk 17]
```diff
@@ -12,22 +12,59 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWavelet.h>" names
...
+                #   Note that this functions returns the pure function value of psi and not the normalized (average=0)
+                #   value given by Psi.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWavelet.pxd:35]:
```suggestion
                #   value given by Psi
```

**Comment by oliveralka** [comment]
@rahul799 Could you please go over the files again and remove the dots from the multi line statements (.pxd only).

**Comment by rahul799** [comment]
Ok. On It.

File: src/pyOpenMS/pxds/IsotopeWaveletTransform.pxd [hunk 18]
```diff
@@ -17,20 +17,80 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/IsotopeWaveletTransform.
...
+                #   Computes the isotope wavelet transform of charge state `c`
+                #   -----
+                #   :param c_trans: The transform
+                #   :param c_ref: The reference spectrum
+                #   :param c: The charge state minus 1 (e.g. c=2 means charge state 3) at which you want to compute the transform
+
         void identifyCharge(MSSpectrum &candidates, MSSpectrum &ref, UInt scan_index, UInt c, double ampl_cutoff, bool check_PPMs) nogil except +
-        void initializeScan(MSSpectrum &c_ref, UInt c) nogil except +
+            # wrap-doc:
+                #   Given an isotope wavelet transformed spectrum @p candidates, this function assigns to every significant
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/IsotopeWaveletTransform.pxd:37]:
```suggestion
                #   Given an isotope wavelet transformed spectrum 'candidates', this function assigns to every significant
```

File: src/pyOpenMS/pxds/JavaInfo.pxd [hunk 19]
```diff
@@ -6,7 +6,15 @@ cdef extern from "<OpenMS/SYSTEM/JavaInfo.h>" namespace "OpenMS":
...
         JavaInfo(JavaInfo &) nogil except +
 
         bool canRun(String java_executable) nogil except +
+            # wrap-doc:
+                #   Determine if Java is installed and reachable
+                #   -----
+                #   The call fails if either Java is not installed or if a relative location is given and Java is not on the search PATH
+                #   -----
+                #   :param java_executable: Path to Java executable. Can be absolute, relative or just a filename
+                #   :param verbose: On error, should an error message be printed to OPENMS_LOG_ERROR?
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/JavaInfo.pxd:19]:
```suggestion
```

Comment (oliveralka) [reply src/pyOpenMS/pxds/JavaInfo.pxd:19]: The parameter `verbose` from the original function (see below) is not wrapped and always true in terms of pyopenms - so this line should be removed. 

`static bool canRun(const String& java_executable, bool verbose_on_error = true); `

File: src/pyOpenMS/pxds/KroenikFile.pxd [hunk 20]
```diff
@@ -8,8 +8,29 @@ from Feature cimport *
 cdef extern from "<OpenMS/FORMAT/KroenikFile.h>" namespace "OpenMS":
 
     cdef cppclass KroenikFile:
+        # wrap-doc:
+                #   File adapter for Kroenik (HardKloer sibling) files
+                #   -----
+                #   The first line is the header and contains the column names:<br>
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/KroenikFile.pxd:14]:
```suggestion
                #   The first line is the header and contains the column names:
```

File: src/pyOpenMS/pxds/LinearInterpolation.pxd [hunk 21]
```diff
@@ -5,26 +5,47 @@ from Types cimport *
...
     cdef cppclass LinearInterpolation[KeyType,ValueType]:
+        # wrap-doc:
+            #   Provides access to linearly interpolated values (and
+            #   derivatives) from discrete data points.  Values beyond the given range
+            #   of data points are implicitly taken as zero.
+            #   -----
+            #   The input is just a vector of values ("Data").  These are interpreted
+            #   as the y-coordinates at the x-coordinate positions 0,...,data_.size-1.
+            #   -----
+            #   The interpolated data can also be <i>scaled</i> and <i>shifted</i> in
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LinearInterpolation.pxd:16]:
```suggestion
            #   The interpolated data can also be scaled and shifted in
```

File: src/pyOpenMS/pxds/LinearInterpolation.pxd [hunk 22]
```diff
@@ -5,26 +5,47 @@ from Types cimport *
...
+            #   the x-dimension by an <em>affine mapping</em>.  That is, we have "inside" and
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LinearInterpolation.pxd:17]:
```suggestion
            #   the x-dimension by an affine mapping.  That is, we have "inside" and
```

File: src/pyOpenMS/pxds/LocalLinearMap.pxd [hunk 23]
```diff
@@ -5,16 +5,30 @@ from Matrix cimport *
 cdef extern from "<OpenMS/ANALYSIS/PIP/LocalLinearMap.h>" namespace "OpenMS":
     
     cdef cppclass LocalLinearMap "OpenMS::LocalLinearMap":
+        # wrap-doc:
+            #   Trained Local Linear Map (LLM) model for peak intensity prediction
+            #   -----
+            #   This class offers a model for predictions of peptide peak heights
+            #   (referred to as intensities) by a Local Linear %Map (LLM) model and
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LocalLinearMap.pxd:12]:
```suggestion
            #   (referred to as intensities) by a Local Linear Map (LLM) model and
```

File: src/pyOpenMS/pxds/LabeledPairFinder.pxd [hunk 24]
```diff
@@ -5,11 +5,16 @@ from BaseGroupFinder cimport *
...
+                #   -----
+                #   Finds feature pairs that have a defined distance in RT and m/z in the same map
+
         # wrap-inherits:
         #  BaseGroupFinder
         LabeledPairFinder() nogil except +
         # private
-        LabeledPairFinder(LabeledPairFinder &) nogil except + # wrap-ignore
-        void run(libcpp_vector[ ConsensusMap ] & input_maps, ConsensusMap & result_map) nogil except +
+        LabeledPairFinder(LabeledPairFinder &) nogil except + #wrap-ignore
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LabeledPairFinder.pxd:17]:
```suggestion
        LabeledPairFinder(LabeledPairFinder &) nogil except + # wrap-ignore
```

File: src/pyOpenMS/pxds/LocalLinearMap.pxd [hunk 25]
```diff
@@ -5,16 +5,30 @@ from Matrix cimport *
...
+            #   A general introduction to the Peak Intensity Predictor (PIP)
+            #   can be found in the PIP Tutorial
+            #   -----
+            #   The model trained needs two files for storing the position of the
+            #   codebook vectors and the linear mappings (codebooks.data, linearMapping.data)
+            #   This is the default model used by PeakIntensityPredictor
+
         LocalLinearMap() nogil except +
-        # private
         LocalLinearMap(LocalLinearMap &) nogil except + # wrap-ignore
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/LocalLinearMap.pxd:23]: Please add the `# private` statement again

Comment (oliveralka) [reply src/pyOpenMS/pxds/LocalLinearMap.pxd:23]: Then it is clear that the copy constructor is private and why we use # wrap-ignore

Comment (oliveralka) [inline src/pyOpenMS/pxds/LocalLinearMap.pxd:23]:
```suggestion
        # private
        LocalLinearMap(LocalLinearMap &) nogil except + # wrap-ignore
```

**Comment by oliveralka** [comment]
rebuild jenkins

**Review by oliveralka — APPROVED** [review]
Looks good!

**Pull request merged**